vmm/src/device_manager/mmio.rs

```diff
@@ -160,15 +226,39 @@ impl MMIODeviceManager {
 
     /// Gets the address of the specified device on the bus.
     pub fn get_address(&self, id: &str) -> Option<&u64> {
-        self.id_to_addr_map.get(id)
+        if let Some(dev_info) = self.id_to_dev_info.get(id) {
+            return Some(&dev_info.addr);
+        }
+        None
     }
+}
+
+/// Private structure for storing information about the MMIO device registered at some address on the bus.
+#[derive(Clone, Debug)]
+pub struct MMIODeviceInfo {
+    addr: u64,
+    irq: u32,
+    len: u64,
+    type_: u32,
+}
 
-    /// Removing the address of a device will generate an error when you try to update the
-    /// drive. The purpose of this method is to test error scenarios and should otherwise
-    /// not be used.
-    #[cfg(test)]
-    pub fn remove_address(&mut self, id: &str) {
-        self.id_to_addr_map.remove(id).unwrap();
+#[cfg(target_arch = "aarch64")]
+impl DeviceInfoForFDT for MMIODeviceInfo {
+    fn addr(&self) -> u64 {
+        self.addr
+    }
+    fn irq(&self) -> u32 {
+        self.irq
+    }
+    fn length(&self) -> u64 {
+        self.len
+    }
+    fn type_(&self) -> FDTDeviceType {
```

Comment:
What do you think about making `type_` in `MMIODeviceInfo` also an `enum`, ideally the same enum as `FDTDeviceType`.

I think you can call this something generic (not fdt related) and define it for both `x86_64` and `aarch64` in order to use it everywhere.

Reply:
Good suggestion. Take a look.